components/kyma-metrics-collector/pkg/process/process.go

```diff
@@ -353,25 +353,75 @@ func isSuccess(status int) bool {
 	return false
 }
 
-func isClusterTrackable(runtime *kebruntime.RuntimeDTO) bool {
-	if runtime.Status.Provisioning != nil &&
-		runtime.Status.Provisioning.State == string(kebruntime.StateSucceeded) &&
-		runtime.Status.Deprovisioning == nil {
+func (p *Process) isClusterTrackable(runtime *kebruntime.RuntimeDTO) bool {
+	// Check if the cluster is in a trackable state
+	trackableStates := map[kebruntime.State]bool{
+		"succeeded": true,
+		"error":     true,
+		"upgrading": true,
+		"updating":  true,
+	}
+
+	if trackableStates[runtime.Status.State] ||
+		(runtime.Status.Provisioning != nil &&
+			runtime.Status.Provisioning.State == string(kebruntime.StateSucceeded) &&
+			runtime.Status.Deprovisioning == nil) {
+
+		// record kebTotalClusters metric
+		recordKEBTotalClusters(
```

Comment:
`isClusterTrackable` is a helper function which called by `populateCacheAndQueue` (which is more of a business logic function), IMO we should track the metrics in `populateCacheAndQueue`.

Also we are resetting the metric in `populateCacheAndQueue`, so it would be good if we manage the metric in this function.

What do you think?

Reply:
Done!